src/Microsoft.AspNetCore.Authentication.OpenIdConnect/OpenIdConnectHandler.cs

```diff
@@ -919,22 +916,18 @@ private string ReadNonceCookie(string nonce)
 
             foreach (var nonceKey in Request.Cookies.Keys)
             {
-                if (nonceKey.StartsWith(OpenIdConnectDefaults.CookieNoncePrefix))
+                if (nonceKey.StartsWith(Options.NonceCookie.Name))
                 {
                     try
                     {
-                        var nonceDecodedValue = Options.StringDataFormat.Unprotect(nonceKey.Substring(OpenIdConnectDefaults.CookieNoncePrefix.Length, nonceKey.Length - OpenIdConnectDefaults.CookieNoncePrefix.Length));
+                        var nonceDecodedValue = Options.StringDataFormat.Unprotect(nonceKey.Substring(Options.NonceCookie.Name.Length, nonceKey.Length - Options.NonceCookie.Name.Length));
                         if (nonceDecodedValue == nonce)
                         {
-                            var cookieOptions = new CookieOptions
+                            var cookieOptions = Options.NonceCookie.Build(Context);
+                            if (Options.NonceCookie.Path == null)
                             {
-                                HttpOnly = true,
-                                Path = OriginalPathBase + Options.CallbackPath,
-                                SameSite = Http.SameSiteMode.None,
-                                Secure = Request.IsHttps
-                            };
-
-                            Options.ConfigureNonceCookie?.Invoke(Context, cookieOptions);
+                                cookieOptions.Path = OriginalPathBase + Options.CallbackPath;
```

Comment:
Hmm. This is problematic. The [default ctor for CookieOptions](https://github.com/aspnet/HttpAbstractions/blob/dev/src/Microsoft.AspNetCore.Http.Features/CookieOptions.cs#L18) always sets Path to "/". If someone implements a custom cookie builder, it will be hard for us to detect if they indend to overwrite the default value or not.

```c#
var builder = new CookieBuilder();
  // builder.Name = null
var options = builder.Build(context)
  // options.Name = "/"

 // builder.Name == null means we can assume the user does not indent to change 
 // the default path, therefore we set it to OriginalPathBase + Options.CallbackPath
```
```c#
var builder = new MyCustomCookieBuilder();
  // builder.Name = null
var options = builder.Build(context)
  // options.Name = "/mypath"

  // builder.Name == null means ????
```